core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/QueryModelNode.java

```diff
@@ -73,16 +73,27 @@
 	/**
 	 * Returns the signature of this query model node. Signatures normally include the node's name and any parameters,
 	 * but not parent or child nodes. This method is used by {@link #toString()}.
-	 * 
+	 *
 	 * @return The node's signature, e.g. <tt>SLICE (offset=10, limit=10)</tt>.
 	 */
 	public String getSignature();
 
 	/**
 	 * Returns a (deep) clone of this query model node. This method recursively clones the entire node tree, starting
 	 * from this nodes.
-	 * 
+	 *
 	 * @return A deep clone of this query model node.
 	 */
 	public QueryModelNode clone();
+
+	/**
+	 * Returns the number of outputs that this QueryNode predicts will be outputted. For a StatementPattern this would
+	 * be the estimated cardinality provided by the EvaluationStatistics. For a Join the would be the resulting number
+	 * of joined tuples.
+	 * 
+	 * @return cardinality
+	 */
+	public double getCardinality();
```

Comment:
Should we call it 'getCostEstimate` as a more generic method? Cost estimation _can_ be done using cardinality, but it's not always the sole factor (for example a SERVICE clause has a significantly higher cost due to network latency).